### ISSUE TITLE
Avoid affecting the theme

### ISSUE THREAD

**Issue by microbit-matt-hillsdon** [rachel-fenichel/blockly-keyboard-experimentation#147]
The keyboard navigation plugin [sets a theme in its constructor](https://github.com/google/blockly-keyboard-experimentation/blob/0402a4d0dae7eeaa24cdd3cfca387af94f87c38d/src/index.ts#L31).

This affects at least the highlight colour in MakeCode:

<img width="503" alt="Image" src="https://github.com/user-attachments/assets/ff3227fa-eea4-4184-aded-4bc1a2032f3f" />

I think @cpcallen previously explained to me that this was temporary / a workaround but I couldn't find anything tracking it so I've raised this issue. If I'm mistaken and it's an accessibility-motivated choice then let's discuss.

**Comment by cpcallen** [comment]
We're planning to introduce some kind of indicator of blocks having focus, and by default moving the cursor will both focus and select the block.

In the meantime, the colour of the cursor (and therefore selection) was changed to (a) match the colour of the cursor marker when positioned in other locations (e.g. on the connection between blocks in a stack), and (b) be more visible—because we had had feedback that the yellow outline was not always very visible (especially on colour picker blocks).

I think the main question of what this will eventually look like should be addressed in the context of focus indicators (perhaps for now in #142, but ultimately in core).

In the mean time: are you requesting that we remove this temporary change now?  We can certainly do it, albeit at the cost of less consistency in the appearance of the cursor.

**Comment by hannahbergam** [comment]
Adding a quick note with another example from Code.org! 

<img width="1702" alt="Image" src="https://github.com/user-attachments/assets/07af41a3-ed21-46c8-9c6f-d948384db005" />

**Comment by samantha-code** [comment]
For prioritization purposes wanted to flag that this bug is blocking Code.org's user testing and getting the new experiment onto our platform. cc: @rachel-fenichel @hannahbergam

**Comment by rachel-fenichel** [comment]
@samantha-code @hannahbergam the offending code is in `src/index.ts`:

```
  /**
   * Update the theme to match the selected glow colour to the cursor
   * colour.
   */
  setGlowTheme() {
    const newTheme = Blockly.Theme.defineTheme('zelosDerived', {
      name: 'zelosDerived',
      base: Blockly.Themes.Zelos,
      componentStyles: {
        selectedGlowColour: '#ffa200',
      },
    });
    this.workspace.setTheme(newTheme);
  }
}
```

Can you check whether you can just set the theme back to your normal theme right after creating/installing keyboard navigation? I believe it would give you a different color highlight but otherwise should behave.

**Comment by hannahbergam** [comment]
> Can you check whether you can just set the theme back to your normal theme right after creating/installing keyboard navigation? I believe it would give you a different color highlight but otherwise should behave.

Yes- I'll check this out!

**Comment by hannahbergam** [comment]
> > Can you check whether you can just set the theme back to your normal theme right after creating/installing keyboard navigation? I believe it would give you a different color highlight but otherwise should behave.
> 
> Yes- I'll check this out!

@rachel-fenichel I was able to reset the theme to work around this! Thanks!

**Comment by microbit-matt-hillsdon** [comment]
Perhaps this should be flipped around so that rather than the keyboard plugin overriding the theme with its preferred glow color:
- the orange color used for keyboard-nav cursor positions could be taken from the theme glow color (e.g. start setting a CSS color variable based on the theme glow color - either in core or the plugin - and use that in relevant CSS)
- separately, folks should be encouraged to use a clear glow color (via docs, maybe changes to defaults over time)

This might be a more reasonable division of responsibilities? The code to hack the theme could then move to the plugin test page.
